tst/unit/test_unit_params.cpp

```diff
@@ -23,24 +23,36 @@
 
 using parthenon::Params;
 
-TEST_CASE("Add and Get is called", "[Add,Get]") {
-  GIVEN("A key") {
+TEST_CASE("Add, Get, and Update are called", "[Add,Get,Update]") {
+  GIVEN("A key with some value") {
     Params params;
     std::string key = "test_key";
     double value = -2.0;
-    params.Add(key, value);
-    double output = params.Get<double>(key);
-    REQUIRE(output == Approx(value));
-    WHEN("parameters are immutable") {
-      auto const &const_params = params;
-      REQUIRE(params.Get<double>(key) == Approx(value));
-    }
-    WHEN("the same key is provided a second time") {
-      REQUIRE_THROWS_AS(params.Add(key, value), std::runtime_error);
-    }
-
-    WHEN("attempting to get the key but casting to a different type") {
-      REQUIRE_THROWS_AS(params.Get<int>(key), std::runtime_error);
+    THEN("we can add it to Params") {
```

Comment:
Should there be a test to make sure it throws if we try to update a non-existent key?

Reply:
Yes, I just added that test.